crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -1856,6 +1983,20 @@ impl<'w> FilteredEntityRef<'w> {
             // SAFETY: We have read access so we must have the component
             .then(|| unsafe { self.entity.get_by_id(component_id).debug_checked_unwrap() })
     }
+
+    /// Gets the component of the given [`TypedComponentId`] from the entity.
+    ///
+    /// You should prefer [`FilteredEntityRef::get`] where possible, this is a niche method
+    /// for accessing dynamic components whose types are known at compile time.
+    pub fn get_dynamic<T: Send + Sync + 'static>(
+        &self,
+        component_id: TypedComponentId<T>,
+    ) -> Option<&'w T> {
+        self.access
+            .has_read(component_id.id)
+            // SAFETY: We have read access so we must have the component.
+            .then(|| unsafe { self.entity.get_dynamic(component_id).debug_checked_unwrap() })
```

Comment:
This is unsound as it doesn't check the `WorldId`.

Reply:
This can be marked as solved (I still don't understand why Github doesn't let the commenter do it...)